### PR TITLE
Fix/rotate mirror

### LINKED ISSUE #389: Build tool remembers "mirror" state between buildings
### Prerequisites 

- [x] I am running the latest **alpha** version of Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.5
- Structurize Version: 0.13.177-alpha
- MineColonies Version (if related bug): 0.14.210-alpha
- Related Mods and their Versions: n/a

## Expected behavior

When you end a build tool session (green checkmark, red x, or either creative mode paste) the build tool forgets whatever the mirror setting was on the next use, like with rotation and positioning.

## Actual behavior

Mirror is remembered, leading to other structures being mirrored (potentially triggering bugs like https://github.com/ldtteam/minecolonies/issues/5006#issuecomment-856323324 or #377)

## Steps to reproduce the issue

1. Start placing a structure and choose to mirror it
2. Either finish placing the structure, or cancel the placement
3. Start placing another structure where you can tell if it's mirrored

### Logs

Logs do not show this issue

### Notes
I bet this is easy to fix but I haven't looked at the build tool code and I need to go to bed tonight instead of fixing it.

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #8719: Rotating/mirroring works wonky (when facing East or West)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.19

### MineColonies Version

minecolonies-1.19.2-1.0.1071-ALPHA

### Structurize Version

structurize-1.19.2-1.0.457-ALPHA

### Related Mods and their Versions

Minecolonies Colony Venture pack, on the patreon server

### Current Behavior

I noticed this first with Shire builder building, but just checked previews of other styles as well, and it is happening there too.
If I bring up the preview of the building, and _first_ rotate it, _then_ mirror it and scroll through each level only level 1 stays in the place I put it
image
This is level one in preview, rotated, then mirrored 

image
This is as soon as I change to level 2

And to add to weirdness, it does not happen when I am facing North or South, only East and West. 

If I mirror it first, then rotate it, the previews all stay in the right place, just the hut block is backwards.

One time I did it the builder DID actually start building it in the wrong position too - I just can't reliably replicate that

### Expected Behavior

Expect it all to look right

### Reproduction Steps

All the steps are above I think! Face east or West, rotate, then mirror, watch weird things happen!

### Logs

https://gist.github.com/Talyda/70336877926ec7ad35c6f6ed27db44ff

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by Nightenom**
Supersedes, closes #592
Closes #389 
Closes ldtteam/minecolonies#8719

### Please test in mcol it actually fixes the issues (did test structurize alone and was working fine)

# Changes proposed in this pull request:
- Bumps forge/mc requirements (1.20 not accepted)
- Replaces renderers cache with guava impl, hopefully won't burn like last time, also slightly adjust renderer hashing
- Introduces RotationMirror in favor of deprecated PlacementSettings 
_(the only useful remnant from struct 2.0 xd)_
- Does NOT replace every usage of PlacementSettings though, will do on next vanilla major

Review please

### CODE REVIEW

File: src/main/java/com/ldtteam/structurize/storage/rendering/types/BlueprintPreviewData.java [hunk 1]
```diff
@@ -246,14 +224,13 @@ private void syncChangesToServer()
 
     /**
      * Move this preview by an offset.
-     * @param pos the offset to move it by.
+     * @param by the offset to move it by.
      */
-    public void move(final BlockPos pos)
+    public void move(final BlockPos by)
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/structurize/storage/rendering/types/BlueprintPreviewData.java]: rather call it offset then

**Review by Raycoms — COMMENTED** [review]
Looks fine overall, did you test this?

**Review by Thodor12 — DISMISSED** [review]
Looks good to me, I'll test it with my specific testcase locally to see if it works and then come back to you

**Comment by uecasm** [comment]
FWIW I have a pending feature on the mcol side where I used a guava blueprint cache and it hard crashed due to OutOfMemory very quickly -- but not in my cache, it crashed in the BlueprintRenderer cache.  I haven't had time to investigate it too closely yet, but it will be interesting to see if this improves it.

File: src/main/java/com/ldtteam/structurize/client/BlueprintHandler.java [hunk 2]
```diff
@@ -139,22 +117,17 @@ public void drawAtListOfPositions(final BlueprintPreviewData previewData,
 
         Minecraft.getInstance().getProfiler().push("struct_render_multi");
```
Comment (uecasm) [inline src/main/java/com/ldtteam/structurize/client/BlueprintHandler.java:120]: javadoc args in this method are outdated

**Review by someaddons — DISMISSED** [review]

File: src/main/java/com/ldtteam/structurize/client/BlueprintHandler.java [hunk 3]
```diff
@@ -139,22 +119,17 @@ public void drawAtListOfPositions(final BlueprintPreviewData previewData,
...
         {
             renderer.draw(previewData, coord, ctx);
         }
 
-        evictTimeCache.put(blueprintHash, System.currentTimeMillis());
-
         Minecraft.getInstance().getProfiler().pop();
     }
+
+    public record RenderingCacheKey(RotationMirror rotationMirror, Blueprint blueprint)
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/structurize/client/BlueprintHandler.java:135]: could use some doc as well and {} same line maybe?

Comment (Nightenom) [reply src/main/java/com/ldtteam/structurize/client/BlueprintHandler.java:135]: we do {} new line, so i keep it that way

**Review by Thodor12 — APPROVED** [review]

**Pull request merged**